tests/backup/backup_resiliency_test.go

```diff
@@ -1558,3 +1558,214 @@ var _ = Describe("{ScaleDownPxBackupPodWhileBackupAndRestoreIsInProgress}", func
+					backupNames = append(backupNames, backupName)
+					wg.Add(1)
+					go func(backupName string, namespace string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						err := CreateBackup(backupName, SourceClusterName, bkpLocationName, backupLocationUID,
+							[]string{namespace}, labelSelectors, orgID, srcClusterUid, "", "", "", "", ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Taking backup %s of application- %s", backupName, namespace))
+					}(backupName, namespace)
+				}
+				backupNamesMap[namespace] = backupNames
+			}
+			wg.Wait()
+			log.Infof("The list of backups taken are: %v", backupNames)
+		})
+
+		Step("Restoring the backed up applications", func() {
+			log.InfoD("Restoring the backed up applications")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				namespaceMapping := make(map[string]string)
+				for _, backupName := range backupNamesMap[namespace] {
+					restoreName := fmt.Sprintf("%s-%s", restoreNamePrefix, backupName)
+					restoreNames = append(restoreNames, restoreName)
+					customNamespace := fmt.Sprintf("new-namespace-%s", RandomString(10))
+					namespaceMapping[namespace] = customNamespace
+					wg.Add(1)
+					go func(restoreName string, backupName string, namespaceMapping map[string]string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						_, err = CreateRestoreWithoutCheck(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Restoring the backup %s with name %s", backupName, restoreName))
+					}(restoreName, backupName, namespaceMapping)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of restores taken are: %v", restoreNames)
+		})
+
+		Step("Cancelling the ongoing restores", func() {
+			log.InfoD("Cancelling the ongoing restores")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, restoreName := range restoreNames {
+				wg.Add(1)
+				go func(restoreName string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					restoreInspectRequest := &api.RestoreInspectRequest{
+						Name:  restoreName,
+						OrgId: orgID,
+					}
+					statusExpected := api.RestoreInfo_StatusInfo_InProgress
+					restoreProgressCheckFunc := func() (interface{}, bool, error) {
+						resp, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
+						if err != nil {
+							return "", false, err
+						}
+						actual := resp.GetRestore().GetStatus().Status
+						if actual == statusExpected {
+							return "", false, nil
+						}
+						if actual == api.RestoreInfo_StatusInfo_Failed {
+							return "", false, fmt.Errorf("restore status for [%s] expected was [%v] but failed", restoreName, actual)
+						}
+						return "", true, fmt.Errorf("restore status for [%s] expected was [%v] but got [%s]", restoreName, statusExpected, actual)
```

Comment:
Since the inspect request's are triggered soon after the creation there is lower chance for restore to be succesful,mostly it will be in retrying in pending state. But yes for safer side will add that to assertion.

Reply:
Cancelling all "running" jobs means job should be inprogress. Otherwise, technically speaking, you cannot say the test was successful, since you are cancelling a job which has finished, right?

Reply:
That true, But here this is the way we can catch inprogress with automation scope from my understanding, or else to avoid the microseconds where restore is succesfull we may need to introduce any network delay points to hold restore in progress for long time.

Reply:
For safer side i'm adding other status in assertion.